src/eduid_webapp/eidas/acs_actions.py

```diff
@@ -188,3 +189,56 @@ def nin_verify_action(session_info, user):
     new_query_string = urlencode({'msg': 'eidas.nin_verify_success'})
     url = urlunsplit((scheme, netloc, path, new_query_string, fragment))
     return redirect(url)
+
+
+@acs_action('mfa-authentication-action')
+@require_user
+def mfa_authentication_action(session_info, user):
+    relay_state = request.form.get('RelayState')
+    current_app.logger.debug('RelayState: {}'.format(relay_state))
+    redirect_url = session.pop(relay_state, None)
```

Comment:
Seems strange and perhaps unsafe to allow the user to pop arbitrary values from the session?